mypy.ini

```diff
@@ -30,15 +30,19 @@ disable_error_code = attr-defined
 [mypy-pkg_resources.tests.*]
 disable_error_code = import-not-found
 
-# - distutils._modified has different errors on Python 3.8 [import-untyped], on Python 3.9+ [import-not-found]
+# - distutils doesn't exist on Python 3.12, unfortunately, this means typing
+#   will be missing for subclasses of distutils on Python 3.12 until either:
+#     - support for `SETUPTOOLS_USE_DISTUTILS=stdlib` is dropped (#3625)
+#       for setuptools to import `_distutils` directly
+#     - or non-stdlib distutils typings are exposed
```

Comment:
This includes potentially moving `_distutils` to a path where the last folder ends in `distutils` so it can be added to the [`MYPY_PATH`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-mypy_path).

Or typeshed accepting to publish a non-stdlib `types-distutils` for Python 3.12+

Until then, Python 3.12+ users won't be able to have complete typing without `types-setuptools` since anything inheriting or re-exported from `distutils` will be typed as `Any`.